credentials/credentials.go

```diff
@@ -208,10 +205,23 @@ func (c *tlsCreds) OverrideServerName(serverNameOverride string) error {
 	return nil
 }
 
+const alpnProtoStrH2 = "h2"
+
+func appendH2ToNextProtos(ps []string) []string {
+	for _, p := range ps {
+		if p == alpnProtoStrH2 {
+			return ps
+		}
+	}
+	ret := make([]string, 0, len(ps)+1)
```

Comment:
This seems like an unnecessary optimisation since the underlying array would at most end up as 4-10 elements in usual cases if the capacity was spent before an append. And this only happens when creating a new transport credential. Out of curiosity: is there something I'm missing?

Reply:
We can't modify the underlying array, as it's owned by the user, so it needs to be copied first.  Since we know the final size of it, there's no reason not to allocate the right amount of memory up front.

Reply:
Thanks for the explanation dfawley, that makes a lot of sense!